Miscellaneous user improvements

GiudGiud opened the pull request:
refs #23791

some things I noticed while running the code

GiudGiud committed:
Output subdomain names on kernel coverage checks, not just subdomain IDs
refs #23791

GiudGiud committed:
Output error on all ranks for missing argument to --split-file

GiudGiud committed:
Move some parameters into parameter groups for the NSFVAction, refs #19472

GiudGiud committed:
Increase timestep tolerance further, as it can still be hit with:
Time Step 1000, time = 10
refs #18819

lindsayad commented:
Initially open pull requests as draft please

GiudGiud committed:
Output subdomain names when checking block restriction of RC UO, refs #23791

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1415678/) on 85e01c1 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/23792/site)

This comment will be updated on new commits.

GiudGiud committed:
Dont compare last timestep on smeared cracking test

That last time step was taking to bypass a rounding error of 1.5e-12.
This is not where we want to be in terms of finishing simulations. We would rather end on 5.999999999999 than take an extra step of up to 1e-12

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1416856/) on 6790949 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>ba109d</th>
      <th colspan=4>#23792 679094</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.87%</td>
      <td>84.87%</td>
      <td>+0.00%</td>
      <td>69.57%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>84193</td>
      <td>84208</td>
      <td>+15</td>
      <td>16</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15009</td>
      <td>15011</td>
      <td>+2</td>
      <td>7</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/23792/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/23792/coverage/framework)


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>ba109d</th>
      <th colspan=4>#23792 679094</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>83.66%</td>
      <td>83.67%</td>
      <td>+0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>10524</td>
      <td>10526</td>
      <td>+2</td>
      <td>2</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>2055</td>
      <td>2055</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/23792/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/23792/coverage/navier_stokes)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/23792/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 69.57% is less than the suggested 90.0%



This comment will be updated on new commits.

GiudGiud commented:
Should be ready now

roystgnr commented on modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C:
```diff
@@ -137,9 +137,12 @@ INSFVRhieChowInterpolator::INSFVRhieChowInterpolator(const InputParameters & par
                  this->name(),
-                 "' doesn't match the block restriction of variable '",
+                 "' (",
+                 std::accumulate(blocks().begin(), blocks().end(), std::string{}),
```
Isn't this just going to concatenate block names without any separating whitespace or punctuation?

GiudGiud replied:
cant remember lemme check

GiudGiud replied:
good catch! not sure what I was thinking

roystgnr reviewed (dismissed):
AssumingIhavemisunderstoodandtheblocknameoutputwon'tlooklikethisIseenootherproblems.

GiudGiud committed:
Address Roy's review
_ use stringify to get a separator between the block names for block restriction error

GiudGiud committed:
Improve formatting of stringify of vectors
space + comma looks better for output
refs #23791

- Adapt tests for new default

roystgnr reviewed (dismissed):
Looks good if CI agrees.  I worry about whether there might be more places that were depending on `stringify()` giving machine-friendly rather than user-friendly output.

GiudGiud commented:
CI wont agree immediately ugh

but I still think this is a natural default for output, while a simple comma is not particularly default for the machine

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1419299/) on 9bb5df2 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23792/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23792/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ba109d0eb9dde4fa0965ce7253c956b13bf00549`

GiudGiud commented:
green green alas
